pkgs/development/python-modules/pylatex/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, matplotlib
+, numpy
+, ordered-set
+, quantities
+}:
+
+buildPythonPackage rec {
+  pname = "pylatex";
+  version = "1.3.4";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "JelteF";
+    repo = "PyLaTeX";
+    rev = "v${version}";
+    hash = "sha256-7JE8xKryf7sjvlUjDH7jUMm7CyirjPbmWo+reWeGIlc=";
+  };
+
+  passthru.optional-dependencies = {
+    matplotlib = [ matplotlib ];
+    matrices = [ numpy ];
+    quantities = [ numpy quantities ];
+  };
+
+  propagatedBuildInputs = [ ordered-set ] ++
```

Comment:
Please avoid adding optional dependencies by default. Users can easily add packages to an environment, but removing packages is non-trivial.

Reply:
Great point!
I'm removing those dependencies so people using the library can just add what they need.